### DISCUSSION TITLE
Server Set-Up - yaml errors

### DISCUSSION THREAD

**Discussion by pepepro2020** [elliotmatson/Docker-Davinci-Resolve-Project-Server#112] [General]
hi there,

i'm trying to configure this davinci server in a docker container on a qnap ts453d. 
probably i'm encountering errors because of the lack of basic know-how about docker, containers, ... SORRY

if i do as recommended in the readme - and do just some very basic adaptions to the provided yaml file like:

---
version: ‘3.8’
x-common:
  database: &db-environment
    POSTGRES_DB: database
    POSTGRES_USER: &pg-user postgres
    POSTGRES_PASSWORD: DaVinci
    TZ: Europe/Vienna
    POSTGRES_LOCATION: &db-location “share/NAS/DavinciServerDatabase:/var/lib/postgresql/data”
  backup: &backup-environment
    SCHEDULE: “@every 1h”
    BACKUP_KEEP_DAYS: 7
    BACKUP_KEEP_WEEKS: 4
    BACKUP_KEEP_MONTHS: 6
    BACKUP_LOCATION: &bk-location “share/NAS//DavinciServerDatabaseBackup:/backups“
  admin: &admin-environment
    PGADMIN_DEFAULT_EMAIL: admin@admin.com
    PGADMIN_DEFAULT_PASSWORD: root
    PGADMIN_PORT: &pgadmin-port "3001:80"
    
    
i get the following error:

Failed to create application "davinciserver". Error message: failed to get image list for yaml: yaml format error: 'services' not found(only support version 2 or 3)

(the locations for postgres and backup are both folders i created on one of my shared volumes called "NAS")

what am i doing wrong?

any help greatfully appreciated

thank you very much!!

**Comment by elliotmatson** [comment]
@pepepro2020 can you add your entire docker-compose.yml in a code block? (minus any passwords or secrets)

**Comment by pepepro2020** [comment]
hi elliot, great to hear from you and many thanks for replying!!

i already found out my first mistake - i just pasted the first part of your yml code - just the first part that one has to edit to ones needs, without all the code down in the "danger zone". (you see, and i'm sorry for that, i'm a complete amateur here) ...

but now, one step ahead, pasting your entire yml code, and that just basically adapted in the first part of the code - as you can see in my following entire docker-compose.yml code block here 

`---
version: '3.8'
x-common:
  database: &db-environment
    POSTGRES_DB: database
    POSTGRES_USER: &pg-user postgres
    POSTGRES_PASSWORD: DaVinci
    TZ: Europe/Vienna
    POSTGRES_LOCATION: &db-location "shares/NAS/DavinciServerDatabase:/var/lib/postgresql/data"
  backup: &backup-environment
    SCHEDULE: "@daily"
    BACKUP_KEEP_DAYS: 7
    BACKUP_KEEP_WEEKS: 4
    BACKUP_KEEP_MONTHS: 6
    BACKUP_LOCATION: &bk-location "shares/NAS//DavinciServerDatabaseBackup:/backups"
  admin: &admin-environment
    PGADMIN_DEFAULT_EMAIL: xxx@gmail.com
    PGADMIN_DEFAULT_PASSWORD: xxx
    PGADMIN_PORT: &pgadmin-port "3001:80"
#
# ------------------------------------------------------------------------------------------
# DANGER ZONE BELOW
#
# The remainder of this file likely does not need to be changed. 
# Please only make modifications
# below if you understand what you are doing.
#
services:
  postgres:
    container_name: resolve_pgsql
    image: postgres:13
    restart: always
    ports:
      - "5432:5432"
    environment:
      <<: [*db-environment]
    volumes:
      - *db-location
    healthcheck:
      test: ["CMD", "pg_isready", "-U", *pg-user]
      interval: 10s
      timeout: 5s
      retries: 5
  pgbackups:
    container_name: resolve_pgbackup
    image: prodrigestivill/postgres-backup-local:13
    restart: always
    volumes:
      - *bk-location
    depends_on:
      - postgres
    environment:
      <<: [*db-environment, *backup-environment]
      POSTGRES_HOST: postgres
      POSTGRES_EXTRA_OPTS: --blobs --format=custom --quote-all-identifiers
      BACKUP_SUFFIX: .backup
      HEALTHCHECK_PORT: 8080
    healthcheck:
      interval: 30s
  pgadmin:
    container_name: resolve_pgadmin
    image: dpage/pgadmin4
    restart: always
    environment:
      <<: [*admin-environment]
    ports:
      - *pgadmin-port
    volumes:
      - pgadmin-config:/pgadmin4-config
      - pgadmin:/var/lib/pgadmin
      - *bk-location
    entrypoint: "/bin/sh"
    command: 
      - -c
      - |
        mkdir -p /var/lib/pgadmin/storage/$${PGADMIN_DEFAULT_EMAIL//@/_}/
        ln -s /backups /var/lib/pgadmin/storage/$${PGADMIN_DEFAULT_EMAIL//@/_}/
        /entrypoint.sh
    depends_on:
      - postgres
      - pgadmin-config
    healthcheck:
      test: ["CMD-SHELL", "wget --no-verbose --tries=1 --spider http://localhost/ || exit 1"]
  pgadmin-config:
    container_name: resolve_pgadmin-config-creator
    image: ghcr.io/elliotmatson/pgadmin-config-creator:latest
    restart: on-failure
    environment:
      <<: [*db-environment]
      PGADMIN_SERVER_JSON_FILE: /pgadmin4-config/servers.json
    volumes:
      - pgadmin-config:/config

volumes:
  pgadmin-config:
  pgadmin:`


... i get the following error from the qnap "container station" respectively from the "create application" action when validating the entered code:

> Invalid Compose YAML Code
> operateApp action [convert] failed: exit status 15: service "postgres" refers to undefined volume shares/NAS/DavinciServerDatabase: invalid compose project

i understand that in my adapted .yml code obviousely the path to the postgres-location and also that one to the backup-location is not correct implemented

but i just don't know how to enter those both paths correctely ...

any suggestions?

any help again greatfully appreciated

THANK YOU

**Reply by pepepro2020** [reply]
hi elliot, me again, i tried and tried and nothing worked ... always errors ... and finally - just for fun and without any serious expectations - asked chatgpt for a check of my yaml code - and chatgpt said that my quotation marks were in the wrong format (i obviousely lost your original ones when adapting the paths to my locations and so had to type them new by myself and used the "curly" ones instead of the "straight" ones - not thinking of doing anything wrong) ...

and with this little intervention IT SUDDENLY WORKED!!!!!

now testing of davinci database server and backup can begin - THANK YOU - if further questions arise i hope i may contact you here again

**Reply by elliotmatson** [reply]
That's great! Let me know if you have any other issues!
